aiida/repository/backend/disk_object_store.py

```diff
@@ -118,3 +118,90 @@ def get_object_hash(self, key: str) -> str:
         if self.container.hash_type != 'sha256':
             return super().get_object_hash(key)
         return key
+
+    def maintain(  # pylint: disable=arguments-differ
+        self,
+        full: bool = False,
+        override_pack_loose: bool = None,
+        override_do_repack: bool = None,
+        override_clean_storage: bool = None,
+        override_do_vacuum: bool = None,
+    ) -> dict:
+        """Performs maintenance operations.
+
+        :param full:
+            a flag to perform operations that require to stop using the maintained profile.
+        :param override_pack_loose:
+            override flag for forcing the packing of loose files.
+        :param override_do_repack:
+            override flag for forcing the re-packing of already packed files.
+        :param override_clean_storage:
+            override flag for forcing the cleaning of soft-deleted files from the repository.
+        :param override_do_vacuum:
+            override flag for forcing the vacuuming of the internal database when cleaning the repository.
+        :return:
+            a dictionary with information on the operations performed.
+        """
+        pack_loose = True
+        do_repack = full
+        clean_storage = full
+        do_vacuum = full
+
+        if override_pack_loose is not None:
+            pack_loose = override_pack_loose
+
+        if override_do_repack is not None:
+            do_repack = override_do_repack
+
+        if override_clean_storage is not None:
+            clean_storage = override_clean_storage
+
+        if override_do_vacuum is not None:
+            do_vacuum = override_do_vacuum
+
+        operation_results = {}
+
+        if pack_loose:
+            self.container.pack_all_loose()
+            operation_results['pack_loose'] = {'decription': 'Packed all loose files.'}
```

Comment:
Maybe it better to have the as logging reports, as you go, rather than only reporting when everything is finished?

```suggestion
            MAINTAIN_LOGGER.report('Packing all loose files.')
            self.container.pack_all_loose()
```